components/ledger/pkg/storage/ledgerstore/migrates/0-init-schema/postgres.sql

```diff
@@ -1,121 +1,133 @@
+create function "VAR_LEDGER_NAME_v2_0_0".meta_compare(metadata jsonb, value numeric, variadic path text[]) returns boolean
+    language plpgsql immutable
+    as $$ begin return jsonb_extract_path(metadata, variadic path)::numeric = value::numeric; exception when others then raise info 'Error Name: %', SQLERRM; raise info 'Error State: %', SQLSTATE; return false; END $$;
 
 --statement
-CREATE FUNCTION "VAR_LEDGER_NAME_v2_0_0".meta_compare(metadata jsonb, value character varying, VARIADIC path text[]) RETURNS boolean
-    LANGUAGE plpgsql IMMUTABLE
-    AS $$ BEGIN return jsonb_extract_path_text(metadata, variadic path)::varchar = value::varchar; EXCEPTION WHEN others THEN RAISE INFO 'Error Name: %', SQLERRM; RAISE INFO 'Error State: %', SQLSTATE; RETURN false; END $$;
+create function "VAR_LEDGER_NAME_v2_0_0".meta_compare(metadata jsonb, value character varying, variadic path text[]) returns boolean
+    language plpgsql immutable
+    as $$ begin return jsonb_extract_path_text(metadata, variadic path)::varchar = value::varchar; exception when others then raise info 'Error Name: %', SQLERRM; raise info 'Error State: %', SQLSTATE; return false; END $$;
 
 --statement
-CREATE FUNCTION "VAR_LEDGER_NAME_v2_0_0".use_account_as_destination(postings jsonb, account character varying) RETURNS boolean
-    LANGUAGE sql
-    AS $_$ select bool_or(v.value::bool) from ( select jsonb_extract_path_text(jsonb_array_elements(postings), 'destination') ~ ('^' || account || '$') as value) as v; $_$;
-
---statement
-CREATE FUNCTION "VAR_LEDGER_NAME_v2_0_0".use_account_as_source(postings jsonb, account character varying) RETURNS boolean
-    LANGUAGE sql
-    AS $_$ select bool_or(v.value::bool) from ( select jsonb_extract_path_text(jsonb_array_elements(postings), 'source') ~ ('^' || account || '$') as value) as v; $_$;
-
---statement
-CREATE FUNCTION "VAR_LEDGER_NAME_v2_0_0".use_account(postings jsonb, account character varying) RETURNS boolean
-    LANGUAGE sql
-    AS $$ SELECT bool_or(v.value) from ( SELECT "VAR_LEDGER_NAME_v2_0_0".use_account_as_source(postings, account) AS value UNION SELECT "VAR_LEDGER_NAME_v2_0_0".use_account_as_destination(postings, account) AS value ) v $$;
-
---statement
-CREATE TABLE IF NOT EXISTS "VAR_LEDGER_NAME_v2_0_0".accounts (
-    address character varying NOT NULL,
-    metadata jsonb DEFAULT '{}'::jsonb,
+create table "VAR_LEDGER_NAME_v2_0_0".accounts (
+    address character varying not null,
+    address_json jsonb not null,
+    metadata jsonb default '{}'::jsonb,
 
     unique(address)
 );
 
 --statement
-CREATE TABLE IF NOT EXISTS "VAR_LEDGER_NAME_v2_0_0".logs_ingestion (
-    onerow_id boolean DEFAULT true NOT NULL,
-    log_id bigint,
-
-    primary key (onerow_id)
-);
-
---statement
-CREATE TABLE IF NOT EXISTS "VAR_LEDGER_NAME_v2_0_0".logs_v2 (
+create table "VAR_LEDGER_NAME_v2_0_0".logs_v2 (
```

Comment:
Keef "if not exists"

Reply:
Useless, it can hide a real problem.
At this step, the table must not exists.